lib/translator/common.cc

```diff
@@ -54,6 +60,36 @@ void SetAllocPageCallbacks(uint64_t (*alloc_callback)(uint16_t),
   dealloc_pages_callback = dealloc_callback;
 }
 
+uint64_t htonll(uint64_t value) {
+  static uint32_t test_val = 42;
+
+  // Check first byte to determine endianness
+  if (*reinterpret_cast<const char*>(&test_val) == test_val) {  // little endian
+    const uint32_t high_bits = htonl(static_cast<uint32_t>(value >> 32));
+    const uint32_t low_bits =
+        htonl(static_cast<uint32_t>(value & 0xFFFFFFFFLL));
+
+    return (static_cast<uint64_t>(low_bits) << 32) | high_bits;
+  } else {  // big endian
+    return value;
+  }
+}
+
+uint64_t ntohll(uint64_t value) {
+  static uint32_t test_val = 42;
+
+  // Check first byte to determine endianness
+  if (*reinterpret_cast<const char*>(&test_val) == test_val) {  // little endian
```

Comment:
If these two functions are the exact same internally should we consider using `"#define ntohll htonll"` for reduced code duplication? I think having both names even if they do the same thing does make sense from a self-documenting standpoint

Reply:
I think using #define is a good idea!